core/blockchain_reader.go

```diff
@@ -330,6 +330,13 @@ func (bc *BlockChain) Validator() Validator {
 	return bc.validator
 }
 
+// SetValidator sets the current validator.
+// This method can be used to force an invalid blockchain to be verified for tests.
+// This method is unsafe and should only be used before block import starts.
+func (bc *BlockChain) SetValidator(v Validator) {
```

Comment:
Perhaps we should move this function to blockchain.go?

This file is basically the collection of all readers.

Reply:
Moved